pkg/controller/util.go

```diff
@@ -134,22 +143,43 @@ Users:
 				case "inrole":
 					t.MemberOf = append(t.MemberOf, s)
 				default:
-					c.logger.Warningf("unknown key %q", p)
+					c.logger.Warningf("unkonwn key %q", p)
 				}
 			}
-
-			delete(data, key)
+			delete(secretData, key)
 		}
 
 		if t.Name != "" {
 			result[t.Name] = t
 		}
 	}
 
-	if len(data) != 0 {
-		c.logger.Warningf("%d unprocessed entries in the infrastructure roles' secret", len(data))
-		c.logger.Info(`infrastructure role entries should be in the {key}{id} format, where {key} can be either of "user", "password", "inrole" and the {id} a monotonically increasing integer starting with 1`)
-		c.logger.Debugf("unprocessed entries: %#v", data)
+	if len(secretData) != 0 {
+		c.logger.Debugf("Extra data in the infrastructure role, checking configmap %v", rolesSecret.Name)
+		// perhaps we have some map entries with usernames, passwords, let's check if we have those users in the configmap
+		if infraRolesMap, err := c.KubeClient.ConfigMaps(rolesSecret.Namespace).Get(rolesSecret.Name, metav1.GetOptions{}); err == nil {
+			// we have a configmap with username - json description, let's read and decode it
+			for role, s := range infraRolesMap.Data {
+				if roleDescr, err := readDecodedRole(s); err != nil {
+					// check if we have a a password in a configmap
+					return nil, fmt.Errorf("could not decode role description: %v", err)
+				} else {
+					c.logger.Debugf("found role description for role %q: %+v", role, roleDescr)
+					if passwd, ok := secretData[role]; ok {
```

Comment:
We sometimes want to get the password (or even the username) from some known key in a different secret that is known to be reliably present in the cluster.  For example, we have to get the ZMON username and password out of this secret: https://github.com/zalando-incubator/kubernetes-on-aws/pull/797

It would be useful to extend the scope of this change a bit so that it allows us to specify infrastructure roles in the ConfigMap but asking for the role name and password to be read from given keys in other secrets.

This would of course require using a more complex struct than `PgUser` for the infrastructure roles ConfigMap values.  I think it's worth doing, though, as we do actually need this right now.

Reply:
Let's do this in acid/database-as-a-service#150